### PR TITLE
refactor: properly name the filter types

### PR DESCRIPTION
**Pull request by fengelniederhammer**

### CODE REVIEW

File: endToEndTests/test/queries/nOf_2of3_aggregated.json [hunk 1]
```diff
@@ -10,17 +10,17 @@
       "matchExactly": false,
       "children": [
         {
-          "type": "NucEq",
+          "type": "NucleotideEquals",
           "position": 241,
           "value": "T"
```
Comment (JonasKellerer) [inline endToEndTests/test/queries/nOf_2of3_aggregated.json:15]: Maybe "symbol" could be a better name?

**Review by JonasKellerer — APPROVED** [review]

**Review by chaoran-chen — COMMENTED** [review]
I am not sure if I find this better. The full names are quite long and `NucMut` is easier to remember for me than `HasNucleotideMutation`. 

@Taepper, what do you think?

**Comment by fengelniederhammer** [comment]
I don't exactly see the point why we should have short, obfuscated names that one has to look up in the documentation. Descriptive names are a key feature for understandable software.
* Long names are a lot more descriptive by themselves and can be understood without reading the docs. To me as a user, it was not clear what `NucEq` actually does as a filter. I only understood it by reading the code.
* Most queries will not be written by hand, i.e. length doesn't matter.
* At least for me, full names are actually easier to remember. If you type a wrong name, SILO should respond with a proper errror message telling you what the allowed values are. I find "remembering" not a problem, because you can simply look it up in the docs or hope for good error messages.

**Comment by chaoran-chen** [comment]
Good points. I don't have a strong opinion and am happy with the change.

One thing regarding `Negated`: should we call it `Not` instead? Because we also write `And` and `Or` (and not something like `Conjunction` and `Disjunction`)

**Comment by fengelniederhammer** [comment]
`Not` is better, I wasn't really happy with `Negated` anyway :)

**Comment by fengelniederhammer** [comment]
@Taepper if you have a different opinion, we can still change it later. We'll merge this so that it doesn't block us.

**Pull request merged**